addons/web/models/ir_http.py

```diff
@@ -36,6 +36,10 @@ def session_info(self):
         mods = odoo.conf.server_wide_modules or []
         lang = user_context.get("lang")
         translation_hash = request.env['ir.translation'].sudo().get_web_translations_hash(mods, lang)
+        home_action_id = user.action_id.id
+        if home_action_id and not self._check_action_exists(home_action_id):
+            user.write({'action_id': False})
+            home_action_id = False
```

Comment:
```suggestion
        home_action = user.action_id
        if home_action:
            home_action = home_action.exists()
            if not home_action:
                user.action_id = False
```

Reply:
as far as i can tell, even:

```py
"home_action_id": user.action_id.exists().id
```

Reply:
That's also an option but I wanted to keep the original's feature that it automatically updates the user's `action_id` when it finds one that's empty. That may be undesirable / unnecessary tho.

Reply:
I do find it odd that in Preferences there is still the Home Action set even if not existing anymore.